test/parser.jl

```diff
@@ -369,7 +369,9 @@ tests = [
         "f() do x, y\n body end"  =>  "(call f (do (tuple x y) (block body)))"
         "f(x) do y body end"  =>  "(call f x (do (tuple y) (block body)))"
         "@f(x) do y body end" =>  "(macrocall-p @f x (do (tuple y) (block body)))"
-
+        "f(x) do (y::T) where T body end"  =>  "(call f x (do (where (parens (::-i y T)) T) (block body)))"
```

Comment:
```suggestion
        "f(x) do y::T where T body end"  =>  "(call f x (do (tuple (::-i y (where T T))) (block body)))"
        "f(x) do (y::T) where T body end"  =>  "(call f x (do (where (parens (::-i y T)) T) (block body)))"
```
Maybe add a test for the case without parenthesis (IIUC the where clause will bind tighter than the ::).